diagnostics-app/src/app/core/services/theme.service.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Global service for managing theme related operations for the app.
+ */
+
+import { EventEmitter, Injectable } from '@angular/core';
+import { Theme } from '../enums/theme.enum';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ThemeService {
+  private _theme: Theme = 'dark';
```

Comment:
Again: I think It doesn't add anything for the type to be Theme rather than string. In fact, you added a new enum file :)

Reply:
I used a type because I was using `'dark' | 'light'` at a lot of places. It checks the value at every place where we use theme and provides a nice auto-complete as well. Thus, using the benefits TypeScript provides.

Reply:
I had a look at TypeScript best practices and updated the project structure accordingly. There's a `types` directory inside `core` and all the types that will be used by various modules/components are added in `global.types.ts`. All the types we create later will reside in this directory.

Similarly when we need `interfaces`, they'll be added in `core/interfaces`.

Reply:
> I used a type because I was using `'dark' | 'light'` at a lot of places.
Can you mention where those places are?

Reply:
Please refer to my comment in global.types.ts.

I think what you are looking for here is Enum and not typealias. By using enums, you can refer to its properties as Theme.DARK or Theme.LIGHT, and forget about using 'dark' and 'light' strings.

Reply:
Yes, I was trying to limit the values. According to multiple typescript sources, [best one](https://blog.bam.tech/developer-news/should-you-use-enums-or-union-types-in-typescript): 

> This means using union types by default, and maybe making an exception if you like to rename things all the time or your tooling doesn’t provide live typecheking and code completion.

A union type seems to be a better fit here since it has to be a string type with limited values that are not going to change anytime soon.